src/renate/memory/storage.py

```diff
@@ -111,6 +129,40 @@ def _set(storage: NestedTensors, idx: int, data_point: NestedTensors) -> None:
         else:
             raise TypeError(f"Expected nested tuple/dict of tensors, found {type(storage)}.")
 
-    def __setitem__(self, idx: int, data_point: NestedTensors) -> None:
-        """Set the item stored at index `idx`."""
-        self._set(self._storage, idx, data_point)
+    def dump_dataset(self, ds):
+        self._length = len(ds)
+        self._storage = self._create_mmap_tensors(self._directory, ds[0], self._length)
+        for idx in range(len(self)):
+            self._set(self._storage, idx, ds[idx])
+
+
+class FileTensorStorage(Storage):
+    """A class implementing permanent storage of nested tensor datasets to disk as pickle files.
+
+    This implements storage for `length` data points consisting of nested tensors of fixed types
+    and shapes. `Storage` implements `__len__` and `__getitem__` and therefore can be used as a
+    torch `Dataset`. To populate the storage, it also implements `dump_dataset`. It does _not_ keep
+    track which slots have or have not been populated.
+
+    `Storage` is given a path to a directory, where it creates (or accesses, if they already exist)
+    pickle files one for each point in the dataset.
+
+    Args:
+        directory: Path to a directory.
+    """
+
+    def __init__(self, directory: str) -> None:
+        super().__init__(directory)
+
+    def dump_dataset(self, ds: torch.utils.data.Dataset) -> None:
+        for i in range(len(ds)):
+            ith_sample = ds[i]
+            save_name = os.path.join(self._directory, f"{i}.pt")
```

Comment:
maybe an overkill, but one could make this a function

Reply:
done.